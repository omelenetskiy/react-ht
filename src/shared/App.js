// Imports
import React, { Fragment } from 'react';
import { Route, Switch } from 'react-router-dom';
import { hot } from 'react-hot-loader/root';
import Loadable from 'react-loadable';

import { ThemeProvider } from 'styled-components';
import AppHeader from './components/header/Header';
import Footer from './components/footer/Footer';
import ErrorBoundary from './components/errorBoundary/ErrorBoundary';

import NotFound from './components/notFound/NotFound';
import Loading from './components/loading/Loading';

import theme from './styled/theme';
import GlobalStyle from './styled/globalStyle';

const Home = Loadable({
  loader: () => import('./components/home/Home'),
  loading: Loading,
});

const MoviePreview = Loadable({
  loader: () => import('./components/moviePreview/MoviePreview'),
  loading: Loading,
});

const App = () => (
  <ThemeProvider theme={theme}>
    <Fragment>
      <GlobalStyle />
      <AppHeader />

      <ErrorBoundary>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/search" component={Home} />
          <Route path="/movie/:id" component={MoviePreview} />
          <Route component={NotFound} />
        </Switch>
      </ErrorBoundary>

      <Footer />
    </Fragment>
  </ThemeProvider>
);

export default hot(App);
