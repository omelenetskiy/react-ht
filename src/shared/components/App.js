// Imports
import React, { Fragment } from 'react';
import { Route, Switch } from 'react-router-dom';
import { hot } from 'react-hot-loader/root';

import Header from './layout/header/Header';
import Footer from './layout/footer/Footer';
import ErrorBoundary from './common/errorBoundary/ErrorBoundary';

import Home from './layout/home/Home';
import MoviePreview from './common/moviePreview/MoviePreview';
import NotFound from './common/notFound/NotFound';

import { ThemeProvider } from 'styled-components';
import { theme } from './styled/theme';
import { GlobalStyle } from './styled/globalStyle';

const App = () => (
  <ThemeProvider theme={theme}>
    <Fragment>
      <GlobalStyle />
      <Header />

      <ErrorBoundary>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/search" component={Home} />
          <Route path="/movie/:id" component={MoviePreview} />
          <Route component={NotFound} />
        </Switch>
      </ErrorBoundary>

      <Footer />
    </Fragment>
  </ThemeProvider>
);

export default hot(App);
