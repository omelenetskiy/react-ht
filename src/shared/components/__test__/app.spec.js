import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-enzyme';
import 'jest-styled-components';

import App from '../App';

describe('App', () => {
  it('should render correctly', () => {
    const context = shallow(<App />);

    expect(toJson(context)).toMatchSnapshot();
  });
});
