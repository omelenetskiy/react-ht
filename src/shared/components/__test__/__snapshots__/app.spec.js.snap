// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "cardMaxWidth": "200px",
      "dangerColor": "#ff4444",
      "dangerColorDark": "#cc0000",
      "infoColor": "#33b5e5",
      "infoColorDark": "#0099cc",
      "maxPageWidth": "1024px",
      "minPageWidth": "360px",
      "stylishColor": "#4b515d",
      "stylishColorDark": "#3e4551",
      "stylishColorLight": "#d2d5da",
      "successColor": "#00c851",
      "successColorDark": " #007e33",
      "warningColor": "#ffbb33",
      "warningColorDark": "#ff8800",
    }
  }
>
  <GlobalStyleComponent />
  <withRouter(Header) />
  <ErrorBoundary>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/search"
      />
      <Route
        component={[Function]}
        path="/movie/:id"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </ErrorBoundary>
  <Footer />
</ThemeProvider>
`;
